### PR TITLE
afform_login_token - Re-import login-token feature as core extension

### PR DESCRIPTION
**Pull request by totten**
Revision History
------------

* Up through 5.78-stable, `afform_core` included support for generating emails  with form-links w/login-tokens. However, there was a long-standing  request from multiple people to support form-links w/page-tokens.

* During 5.79-alpha (#30585), we introduced page-tokens. We discussed more  at the sprint, and there was a general feeling that page-tokens were more  desirable (more appropriate to more users/less foot-gunny). We agreed that  login-tokens were hypothetically better for some, but no one in that  discussion was eager to use them. So we moved login-tokens to a contrib  extension.

* But during 5.79-beta cycle, we got testing feedback from more people who were keen to keep login-tokens in a more accessible place.

* This PR re-introduces login-token support as a core-extension for 5.79-beta,  which means that:

    * The default/typical mode of operation is based on page-tokens.
    * Using login-tokens is a little bit of work, but not as much.

Before
---------

Upgrader suggests you download a contrib extension. The extension has a README which talks about migration.

Screenshot from 2024-10-29 22-28-51

After 
-------

Extension is included. Upgrader suggests reading the upgrade notice.

Screenshot from 2024-10-29 22-28-33

(*At the moment, I'm working on the notice and pulling from the earlier README as source-material...*)

### CODE REVIEW

**Comment by totten** [comment]
* I'd usually be apprehensive about a change of this size at this point in RC -- however, the functionality existed before, and the alternative (adding it back via 5.80 or 5.81) would be painfully bouncy for sysadmins.

* When adding core-ext, it's good to check effect on distmaker. Flagging `run-distmaker` so we can double-check that.

**Comment by eileenmcnaughton** [comment]
@totten thanks for this - is it possible to put the 'notice' in the sysadmin guide? I feel like that would be a better permanent place - although if some of the detail is time specific it could point from there to the wiki?

**Comment by eileenmcnaughton** [comment]
@totten I've added merge-ready cos I think my comment is non-blocking (easy to change the link if we choose) & you should merge this when it makes sense to you

**Comment by ufundo** [comment]
Looks fine to me :+1:

File: ext/afform/login_token/info.xml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    <author>
+      <name>CiviCRM</name>
+      <email>info@civicrm.org</email>
+      <role>Maintainer</role>
+    </author>
+  </authors>
+  <urls>
+    <url desc="Main Extension Page">https://civicrm.org</url>
+    <url desc="Documentation">https://lab.civicrm.org/extensions/afform_login_token</url>
+    <url desc="Support">https://lab.civicrm.org/extensions/afform_login_token</url>
```
Comment (ufundo) [inline ext/afform/login_token/info.xml:17]: This URL is a 404 now?

**Comment by ufundo** [comment]
Have now tested this by upgrading a 5.78 Standalone site using the generated tarball, and I got the upgrade notice and was able to install the extension and then select `Message Tokens (Login)`

image


In line with @eileenmcnaughton suggestion, I think it would be good to have the guidance in the Docs rather than the Wiki. I'd prioritise "here's how it now works" rather than the specific guidance for upgrading (it would be nice to have both, but time is short etc). But not a blocker on merging here.

File: ext/afform/login_token/info.xml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  <releaseDate>2024-09-18</releaseDate>
+  <version>1.0.0</version>
+  <develStage>alpha</develStage>
+  <compatibility>
+    <ver>5.79</ver>
+  </compatibility>
+  <requires>
+    <ext>org.civicrm.afform</ext>
+  </requires>
+  <comments>This is a new, undeveloped module</comments>
```
[the remark this answers is not in the document]
Comment (totten) [reply ext/afform/login_token/info.xml]: Right-o. Updated.

**Comment by totten** [comment]
Thanks @ufundo @eileenmcnaughton

**Pull request merged**